CMakeLists.txt

```diff
@@ -665,17 +665,64 @@ endif()
 # the zephyr elf, we set the kernel_elf file name to zephyr.elf.
 set_target_properties(${logical_target_for_zephyr_elf} PROPERTIES OUTPUT_NAME ${KERNEL_NAME})
 
-add_custom_command(
-  TARGET ${logical_target_for_zephyr_elf}
-  POST_BUILD
+set(post_build_commands "")
+
+# TODO: Write a KConfig for these options instead
+set(CONFIG_BUILD_OUTPUT_CHECK_LINK_MAP 1)
+set(CONFIG_BUILD_OUTPUT_HEX 1)
+set(CONFIG_BUILD_OUTPUT_BIN 1)
+set(CONFIG_BUILD_OUTPUT_S19 1)
+set(CONFIG_BUILD_OUTPUT_DISASSEMBLY 1)
+set(CONFIG_BUILD_OUTPUT_STAT 1)
+set(CONFIG_BUILD_OUTPUT_STRIPPED 1)
+
+list_append_ifdef(CONFIG_BUILD_OUTPUT_CHECK_LINK_MAP
+  post_build_commands
   COMMAND ${PYTHON_EXECUTABLE} $ENV{ZEPHYR_BASE}/scripts/check_link_map.py ${KERNEL_MAP_NAME}
```

Comment:
This is not an output, this is just a verification script that we run on every build, so I guess it can be added by default. Maybe just change CONFIG_BUILD_OUTPUT_CHECK_LINK_MAP to CONFIG_CHECK_LINK_MAP and enable by default here (we can add a Kconfig that is enabled by default at some point)

Reply:
Fixed

Reply:
thanks, I am already building on top of that PR :)